mlflow/server/js/src/model-registry/components/ModelPage.tsx

```diff
@@ -48,79 +70,263 @@ type ModelPageImplProps = WithRouterNextProps<{ subpage: string }> & {
+      urlParams['page'] = page;
+    }
+    const newUrl = createMLflowRoutePath(
+      `${ModelRegistryRoutes.getModelPageRoute(this.props.modelName)}?${Utils.getSearchUrlFromState(urlParams)}`,
+    );
+    if (newUrl !== this.props.location.pathname + this.props.location.search) {
+      this.props.navigate(newUrl);
+    }
+  };
+
+  resetHistoryState() {
+    this.setState((prevState: any) => ({
+      currentPage: 1,
+      pageTokens: this.defaultPersistedPageTokens,
+    }));
+    this.setPersistedPageTokens(this.defaultPersistedPageTokens);
+  }
+
+  static getOrderByExpr = (orderByKey: any, orderByAsc: any) =>
+    orderByKey ? `${orderByKey} ${orderByAsc ? 'ASC' : 'DESC'}` : '';
+
+  isEmptyPageResponse = (value: any) => {
+    return !value || !value.model_versions || !value.next_page_token;
+  };
+
+  // Loads the initial set of model versions.
+  loadModelVersions(isInitialLoading = true) {
+    this.loadPage(this.state.currentPage, isInitialLoading, true);
+  }
+
+  /**
+   * Returns a LocalStorageStore instance that can be used to persist data associated with the
+   * ModelRegistry component.
+   */
+  static getLocalStore(key: any) {
+    return LocalStorageUtils.getSessionScopedStoreForComponent('ModelPage', key);
+  }
+
+  getPersistedPageTokens() {
+    const store = ModelPageImpl.getLocalStore(this.modelPageStoreKey);
+    if (store && store.getItem('page_tokens')) {
+      return JSON.parse(store.getItem('page_tokens'));
+    } else {
+      return this.defaultPersistedPageTokens;
+    }
+  }
+
+  setPersistedPageTokens(page_tokens: any) {
+    const store = ModelPageImpl.getLocalStore(this.modelPageStoreKey);
+    if (store) {
+      store.setItem('page_tokens', JSON.stringify(page_tokens));
+    }
+  }
+
+  getPersistedMaxResults() {
+    const store = ModelPageImpl.getLocalStore(this.modelPageStoreKey);
+    if (store && store.getItem('max_results')) {
+      return parseInt(store.getItem('max_results'), 10);
+    } else {
+      return MODEL_VERSIONS_PER_PAGE_COMPACT;
+    }
+  }
+
+  setMaxResultsInStore(max_results: any) {
+    const store = ModelPageImpl.getLocalStore(this.modelPageStoreKey);
+    store.setItem('max_results', max_results.toString());
+  }
+
   handleEditDescription = (description: any) => {
     const { model } = this.props;
     return this.props
       .updateRegisteredModelApi(model.name, description, this.updateRegisteredModelApiId)
-      .then(this.loadData);
+      .then(() => this.loadPage(1, false, true));
   };
 
   handleDelete = () => {
     const { model } = this.props;
-    return this.props.deleteRegisteredModelApi(model.name, this.deleteRegisteredModelApiId);
+    return this.props.deleteRegisteredModelApi(model.name, this.deleteRegisteredModelApiId).then(() => {
+      this.props.navigate(ModelRegistryRoutes.modelListPageRoute);
+    });
   };
 
-  loadData = (isInitialLoading: any) => {
+  loadPage = (page: any, isInitialLoading: any, loadModelMetadata = false) => {
```

Comment:
are these the right types?
```suggestion
  loadPage = (page: number, isInitialLoading: boolean, loadModelMetadata = false) => {
```